apps/retrospective/services/sprint_task.go

```diff
@@ -132,6 +132,48 @@ func (service SprintTaskService) Update(sprintTaskID string, retroID string, spr
 	return service.Get(sprintTaskID, retroID, sprintID)
 }
 
+// Delete ...
+func (service SprintTaskService) Delete(sprintTaskID string, retroID string, sprintID string) (int, error) {
+	gormDB := service.DB
+
+	var sprintTask retroModels.SprintTask
+	err := gormDB.Model(retroModels.SprintTask{}).Where("id = ?", sprintTaskID).Scan(&sprintTask).Error
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	var sprintTasksCount int
+	err = gormDB.Model(&retroModels.SprintTask{}).Where("task_id = ?", sprintTask.TaskID).Count(&sprintTasksCount).Error
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	// delete sprintTask.
+	err = gormDB.Where("id = ?", sprintTaskID).Delete(retroModels.SprintTask{}).Error
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+	// delete sprintMemberTask
+	err = gormDB.Where("sprint_task_id = ?", sprintTaskID).Delete(retroModels.SprintMemberTask{}).Error
```

Comment:
Shouldn't we delete SMTs first as Sprint task is used in those SMTs?

Reply:
done.